### ISSUE TITLE
Illegal Instruction Error on latest 2.8.1 image

### ISSUE THREAD

**Issue by rtthane** [jc21/nginx-proxy-manager#1012]
Checklist

Have you pulled and found the error with jc21/nginx-proxy-manager:latest docker image? yes
Are you sure you're not using someone else's docker image? yes
If having problems with Lets Encrypt, have you made absolutely sure your site is accessible from outside of your network? not relevant

Describe the bug

Have the following error in the docker container logs:

nginx line 18: 1631 Illegal instruction (core dumped) node --abort on uncaught exception --max_old_space_size=250 index.js

Image will now no longer start. Error started a few hours ago.

What version of Nginx Proxy Manager is reported on the login page?
Cannot get to login page

**Comment by TheChrisK** [comment]
Same issue here after upgrading to 2.8.1. Is there a stable version I can safely downgrade to?

**Comment by rtthane** [comment]
I have been on 2.8.1 since it was originally released (28 days ago according to Github) which was fine, but it seems that the 2.8.1 image was modified and pushed a few hours ago which is what seems to be causing this (for me anyway)

**Comment by TheChrisK** [comment]
> 
> 
> I have been on 2.8.1 since it was originally released (26 days ago according to Github) which was fine, but it seems that the 2.8.1 image was modified and pushed a few hours ago which is what seems to be causing this (for me anyway) - my Watchtower auto pulled it this morning.

My watchtower did too. I downgraded to 2.8.0 but receiving the following error:

`
[4/12/2021] [3:33:38 PM] [Migrate  ] › ℹ  info      Current database version: 20210210154703


[4/12/2021] [3:33:38 PM] [Global   ] › ✖  error     The migration directory is corrupt, the following files are missing: 20210210154702_redirection_scheme.js, 20210210154703_redirection_status_code.js`

Both versions are unusable for me at this point.

**Comment by rtthane** [comment]
> **Edit:** Replaced the missing files from the new version and downgraded to 2.8.0 and it seems to be working now. I can access the front end and the proxy appears functional for now.

Fortunately I had a full backup with I reverted to which got me working again. Made sure I stopped Watchtower for now though until resolved :-)

**Comment by freender** [comment]
Lets document workaround: 
1) ``vi docker-compose.yml``
and change label to 'jc21/nginx-proxy-manager:2.8.0'

2) Pull new image
``
docker-compose pull
``
3) Create migration files
``
docker exec -it nginx_proxy_manager /bin/sh
``
cd /app/migrations
touch 20210210154702_redirection_scheme.js
touch 20210210154703_redirection_status_code.js
``
4) Restart docker container
docker-compose restart nginx_proxy_manager

**Comment by TheChrisK** [comment]
> Let's document workaround:
> 
>     1. `vi docker-compose.yml`
>        and change label to 'jc21/nginx-proxy-manager:2.8.0'
> 
>     2. Pull new image
>        `docker-compose pull`
> 
>     3. Create migration files
>        `docker exec -it nginx_proxy_manager /bin/sh`
>        `cd /app/migrations`
>        `touch 20210210154702_redirection_scheme.js`
>        `touch 20210210154703_redirection_status_code.js`
> 
>     4. Restart docker container
>        `docker-compose restart nginx_proxy_manager`

Thank you. I actually had to copy the files entirely as the blank .js files didn't work to completely get me up and running..

**Comment by freender** [comment]
Additional Logs:

```
pi@pi-proxy:~/docker $ docker logs nginx_proxy_manager
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01_s6-secret-init.sh: executing...
[cont-init.d] 01_s6-secret-init.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
❯ Enabling IPV6 in hosts: /etc/nginx/conf.d
  ❯ /etc/nginx/conf.d/production.conf
  ❯ /etc/nginx/conf.d/include/proxy.conf


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbed0219c
  ❯ /etc/nginx/conf.d/include/ip_ranges.conf
  ❯ /etc/nginx/conf.d/include/assets.conf
  ❯ /etc/nginx/conf.d/include/ssl-ciphers.conf
  ❯ /etc/nginx/conf.d/include/force-ssl.conf
  ❯ /etc/nginx/conf.d/include/letsencrypt-acme-challenge.conf
  ❯ /etc/nginx/conf.d/include/block-exploits.conf
  ❯ /etc/nginx/conf.d/include/resolvers.conf
  ❯ /etc/nginx/conf.d/default.conf
./run: line 18:   223 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js
❯ Enabling IPV6 in hosts: /data/nginx
  ❯ /data/nginx/proxy_host/18.conf
  ❯ /data/nginx/proxy_host/15.conf
  ❯ /data/nginx/proxy_host/11.conf
  ❯ /data/nginx/proxy_host/1.conf
  ❯ /data/nginx/proxy_host/14.conf
  ❯ /data/nginx/proxy_host/8.conf
  ❯ /data/nginx/proxy_host/22.conf
  ❯ /data/nginx/proxy_host/2.conf
  ❯ /data/nginx/proxy_host/13.conf
  ❯ /data/nginx/proxy_host/16.conf
  ❯ /data/nginx/proxy_host/3.conf
  ❯ /data/nginx/proxy_host/19.conf
  ❯ /data/nginx/proxy_host/5.conf
  ❯ /data/nginx/proxy_host/4.conf
  ❯ /data/nginx/proxy_host/12.conf
  ❯ /data/nginx/proxy_host/23.conf
  ❯ /data/nginx/proxy_host/9.conf
  ❯ /data/nginx/proxy_host/6.conf
  ❯ /data/nginx/proxy_host/7.conf
  ❯ /data/nginx/custom/http.conf
  ❯ /data/nginx/default_host/site.conf


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbe83719c
./run: line 18:   288 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbeab219c
./run: line 18:   295 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbe9d219c
./run: line 18:   302 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbe8f319c
./run: line 18:   309 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbe88819c
./run: line 18:   326 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbe99519c
./run: line 18:   333 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbeaed19c
./run: line 18:   340 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbe96019c
./run: line 18:   347 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbeda119c
./run: line 18:   354 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbee3019c
./run: line 18:   366 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbea2d19c
./run: line 18:   377 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbe9f719c
./run: line 18:   384 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js


#
# Fatal error in , line 0
# unreachable code
#
#
#
#FailureMessage Object: 0xbe86519c
./run: line 18:   391 Illegal instruction     (core dumped) node --abort_on_uncaught_exception --max_old_space_size=250 index.js
```

**Comment by jc21** [comment]
2.8.1 was rebuilt yesterday to address a vulnerability in openssl. This problem is likely due to the base nginx image: https://hub.docker.com/r/jc21/alpine-nginx-full.

To investigate if this image will run correctly on your architecture:

```bash
docker run --rm -ti jc21/alpine-nginx-full nginx -V
```

Should this NOT work for you, **please mention the architecture you're running it on**. Someone has already mentioned arm7 (thank you!)

I'll continue to investigate what's happened here, but ..

### For now
To go back to a previous, but compatible build, use the `github-develop` docker tag.

**Comment by jc21** [comment]
Ok looks like it's nodejs then! I'll have it fixed hopefully today

**Comment by TheChrisK** [comment]
> 
> 
> Ok looks like it's nodejs then! I'll have it fixed hopefully today

Thank you for your prompt response to this. I switched to `github-develop` and so far so good.

**Comment by rtthane** [comment]
> Ok looks like it's nodejs then! I'll have it fixed hopefully today

Many thanks for looking into this so promptly for us.

**Comment by DarkVentHead** [comment]
Just wanted to pop by as i'm having the same error as well. Currently in process of downloading github-develop tag as well to see if it works for me. Much appreciated to the quick response time!

**Comment by Stan-Gobien** [comment]
Thanks for the github-develop tag. Going to try that now. Armv7 arch (Pi4).

**Comment by jc21** [comment]
So the cause of the problem is this: nodejs packages in alpine have been updated since the last time the base image was built. Unfortunately they are broken for arm7, at least.

I can't seem to find a fixable workaround for it, so instead I've repushed the `github-develop` branch to `latest`, `2`, `2.8.1`. This problem unfortunately affects all future releases as well.

I'll close the issue as those images should be fine now.
